### PR TITLE
Add support for ALB ingress using annotations

### LINKED ISSUE #987: ALB Ingress Controller "use-annotation" backends
**What steps did you take and what happened:**

AWS ALB Ingress Controller allows you to reference an annotation as a Ingress Backend.ServiceName like in this example https://kubernetes-sigs.github.io/aws-alb-ingress-controller/guide/tasks/ssl_redirect/

If you apply that yaml example and try navigating to the Ingress view the following error is thrown:

```
2020-06-12T23:47:41.177-0400	ERROR	api/content_manager.go:130	generate content	{"client-id": "9cea0b24-ad28-11ea-ba16-acde48001122", "err": "generate content: add row for object: get status for object: build status for ingress: get service from object store: get object from object store: services \"ssl-redirect\" not found", "content-path": "overview/namespace/default/discovery-and-load-balancing/ingresses"}
github.com/vmware-tanzu/octant/internal/api.(*ContentManager).runUpdate.func1
	/private/tmp/octant-20200604-13237-14eiloi/src/github.com/vmware-tanzu/octant/internal/api/content_manager.go:130
github.com/vmware-tanzu/octant/internal/api.(*InterruptiblePoller).Run.func1
	/private/tmp/octant-20200604-13237-14eiloi/src/github.com/vmware-tanzu/octant/internal/api/poller.go:86
github.com/vmware-tanzu/octant/internal/api.(*InterruptiblePoller).Run
	/private/tmp/octant-20200604-13237-14eiloi/src/github.com/vmware-tanzu/octant/internal/api/poller.go:95
github.com/vmware-tanzu/octant/internal/api.(*ContentManager).Start
	/private/tmp/octant-20200604-13237-14eiloi/src/github.com/vmware-tanzu/octant/internal/api/content_manager.go:106
```

This also affects the "Resource Viewer" as if you view a service "Resource Viewer" it will try to parse the ingress that the service is attached to and will fail finding this service.

**What did you expect to happen:**

The service to be ignored and no errors found when viewing the Resource Viewer.

**Anything else you would like to add:**

Am happy to create a PR for this, I can see that backends are looked up in these 2 places:

* https://github.com/vmware-tanzu/octant/blob/master/internal/objectstatus/ingress.go#L177
* https://github.com/vmware-tanzu/octant/blob/master/internal/queryer/queryer.go#L405

It could just skip the backend `if ServicePort == "use-annotation"`.

And/Or it could `return nil, nil` if the error is NotFound from the DynamicCache. However Octant will still mark the Ingress object status has having an error as it expects the backend to be there unless we implement the first option as well.

**Environment:**

- Octant version (use `octant version`): 0.13.1
- Kubernetes version (use `kubectl version`): v1.18.3
- OS (macOS 10.15, Windows 10, Ubuntu 19.10 etc): macOS 10.15

### PR DESCRIPTION
**Pull request by GuessWhoSamFoo**
**What this PR does / why we need it**:
~AWS ALB Ingress controller have references to a service through annotations. Rather than writing ALB-specific logic in Octant, we can have the resource viewer and object status tell the user that the service cannot be found and offload more detailed views via plugins.~

This PR checks validates the action in the annotations if the service port is `use-annotation`.

**Which issue(s) this PR fixes**
- Fixes #987 

Signed-off-by: GuessWhoSamFoo <foos@vmware.com>

### CODE REVIEW

**Comment by wwitzel3** [comment]
Excellent, thank you @GuessWhoSamFoo

**Comment by wwitzel3** [comment]
So taking https://kubernetes-sigs.github.io/aws-alb-ingress-controller/guide/tasks/ssl_redirect/ as the example, we'd want the resource viewer to be green (assuming both `user-service` and `default-service` existed)

**Comment by wwitzel3** [comment]
So taking https://kubernetes-sigs.github.io/aws-alb-ingress-controller/guide/tasks/ssl_redirect/ as the example, we'd want the resource viewer to be green (assuming both `user-service` and `default-service` existed)

**Comment by wwitzel3** [comment]
What if we checked if a servicePort was an int? And if it is not an int, we can add that backend and it's service port under the Ingress config? This is a more general approach and won't require a plugin.

**Comment by GuessWhoSamFoo** [comment]
The PR is updated specifically to handle the non-int case specifically for a serviceName to match the action in an annotation. Added code has no impact on other users and the ALB ingress controller is supported by a k8s sig so it should be fine being included in octant core until these cases are encountered more frequently.

File: internal/objectstatus/ingress.go [hunk 1]
```diff
@@ -74,6 +75,15 @@ func (is *ingressStatus) run(ctx context.Context) (ObjectStatus, error) {
 	}
 
 	for _, backend := range backends {
+		if backend.ServicePort.String() == "use-annotation" {
+			albAction := ingressAlbActionAnnotation + backend.ServiceName
+			if _, ok := ingress.Annotations[albAction]; !ok {
+				status.SetError()
+				status.AddDetailf("Backend refers to annotations %q which does't exist", albAction)
```
Comment (wwitzel3) [inline internal/objectstatus/ingress.go:82]: I think we should grab the information out of the annotations and write it to the config card.

**Pull request merged**